presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -191,19 +195,24 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
     {
         InformationSchemaTableHandle table = (InformationSchemaTableHandle) handle;
 
-        if (!isTablesEnumeratingTable(table.getTable()) || !table.getPrefixes().equals(defaultPrefixes(catalogName))) {
-            return Optional.empty();
+        Optional<Set<String>> roles = table.getRoles();
+        Optional<Set<String>> grantees = table.getGrantees();
+        if (ROLE_AUTHORIZATION_DESCRIPTORS.equals(table.getTable()) && table.getRoles().isEmpty() && table.getGrantees().isEmpty()) {
+            roles = calculateRoles(session, constraint.getSummary(), constraint.predicate());
+            grantees = calculateGrantees(session, constraint.getSummary(), constraint.predicate());
         }
 
-        Set<QualifiedTablePrefix> prefixes = getPrefixes(session, table, constraint);
-
-        if (prefixes.equals(table.getPrefixes())) {
-            return Optional.empty();
+        Set<QualifiedTablePrefix> prefixes = table.getPrefixes();
+        if (isTablesEnumeratingTable(table.getTable()) && table.getPrefixes().equals(defaultPrefixes(catalogName))) {
```

Comment:
I don't think we should do such ifs. I would prefer to try to update `prefixes` and `grantees` always. And if any of them is not changed return `Optional.empty()` (as it was before).

Reply:
Agreed. Much better.

Reply:
Done.